source/grid/manifold_lib.cc

```diff
@@ -953,8 +964,9 @@ namespace
         // in 2D but it becomes clear in 3D where we avoid looking at the faces'
         // orientation and other complications).
         double weights_vertices[GeometryInfo<2>::vertices_per_cell];
-        for (unsigned int v=0; v<GeometryInfo<2>::vertices_per_cell; ++v)
-          weights_vertices[v] = -GeometryInfo<2>::d_linear_shape_function(chart_point, v);
+        for (unsigned int i1=0, v=0; i1<2; ++i1)
+          for (unsigned int i0=0; i0<2; ++i0, ++v)
+            weights_vertices[v] = -linear_shapes[2+i1] * linear_shapes[i0];
```

Comment:
If I have understood things correctly then these weights include a `-1` factor and we use `+=` later, but in 3D we don't have a `-1` factor and we do `-=` later. If I am correct, then would you mind using the same convention for both?

If we go with the positive convention then we can use the values in `weights_vertices` to calculate `new_point`: a further (small) saving.

Reply:
It took me a little while to understand it but now I think I got you. Please have a look at the new code when I commit it: I still need to fix a bug and have to rush away now, so that will happen later today.

Reply:
This looks consistent now.